addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+   * and color.
+   * @type {Array.<Object>}
+   */
+  finishedData: Ember.computed('sortedData', 'otherSliceLabel', function() {
+    var posTop, negBottom, stackedValues, otherSliceLabel;
+    otherSliceLabel = this.get('otherSliceLabel');
+    return _.map(this.get('sortedData'), (values, barLabel) => {
+      posTop = 0;
+      negBottom = 0;
+      stackedValues = _.map(values, function(d) {
+        var yMin, yMax;
+        if (d.value < 0) {
+          yMax = negBottom;
+          negBottom += d.value;
+          yMin = negBottom;
+        } else {
+          yMin = posTop;
+          posTop += d.value;
+          yMax = posTop;
+        }
+        return {
+          yMin: yMin,
+          yMax: yMax,
+          value: d.value,
+          barLabel: d.barLabel,
+          sliceLabel: d.sliceLabel,
+          color: d.color
+        };
+      });
+
+      return {
+        barLabel: barLabel,
+        values: values,
+        stackedValues: stackedValues,
+        max: posTop,
+        min: negBottom
+      };
+    });
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Slice and Bar Sorting
+  // ---------------------------------------------------------------------------
+
+  /**
+   * Key used to determine slice sorting order. Can be 'value', 'original', or
+   * 'other'.
+   * @see valueSliceSortingFn
+   * @see originalSliceSortingFn
+   * @see customSliceSortingFn
+   * @type {string}
+   */
+  sliceSortKey: 'value',
+
+  /**
+   * Slice order for when sliceSortKey is set to `value`
+   * Starting with the largest net-value bar, sort slices in each bar by abs.
+   * value, and add these to the slice order (from largest to smallest)
+   * assuming they are not already in the order. Then repeat this process for
+   * all bars to make sure all slices are listed in the order.
+   * @see sliceSortKey
+   * @see valueSliceSortingFn
+   * @type {Array.<string>}
+   */
+  sliceOrderByValue: Ember.computed('netBarValues.[]', 'dataGroupedByBar', 'otherSliceLabel', function() {
+    var sortedBars, sliceOrder, slicesInBar, allSlicesByBar;
+    allSlicesByBar = this.get('dataGroupedByBar');
+    sortedBars = _.sortBy(this.get('netBarValues'), 'value').reverse();
+    sliceOrder = [];
+    sortedBars.forEach(bar => {
+      slicesInBar = _.sortBy(allSlicesByBar[bar.barLabel], slice => {
+        return -Math.abs(slice.value);
+      });
+      slicesInBar.forEach(slice => {
+        if (sliceOrder.indexOf(slice.sliceLabel) === -1) {
+          sliceOrder.push(slice.sliceLabel);
+        }
+      });
+    });
+    sliceOrder.push(this.get('otherSliceLabel'));
+    return sliceOrder;
+  }),
+
+  /**
+   * Comparison function for slices for when sliceSortKey is 'value'
+   * @see sliceSortKey
+   * @see sliceOrderByValue
+   * @type {function}
+   */
+  valueSliceSortingFn: Ember.computed('sliceOrderByValue.[]', function() {
+    var sliceOrder = this.get('sliceOrderByValue');
```

Comment:
`sliceOrderByValue` is a global slice sorting order for EVERY bar. Within this comparison function, we're only looking at slices within one bar. When sorting by value, we want a consistent slice order across all bars -- to do this, we give higher weight to slices in the highest net value bar, then to the second highest bar, etc. The full details on how this is calculated are in the spec (as well as PR description, I believe).

Example: 
`Bar 1` is big bar, `Bar 2` is small bar. 
In `Bar 1`, we have `Slice A` = 10, `Slice B` = 5. 
In `Bar 2`, we have `Slice A` = 1, `Slice B` = 5. 
If we sort by value, BOTH bars should sort `Slice A` ahead of `Slice B`, because `Slice A` is bigger than `Slice B` in the tall bar.
TL;DR -- slice order must be the same across all bars, so we can't just do a simply sort slices within a bar by value.